src/Command/Argument/Search/CreateArguments.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/*
+ * This file is part of the Predis package.
+ *
+ * (c) 2009-2020 Daniele Alessandri
+ * (c) 2021-2023 Till Krüss
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Predis\Command\Argument\Search;
+
+use InvalidArgumentException;
+use Predis\Command\Argument\ArrayableArgument;
+
+class CreateArguments implements ArrayableArgument
+{
+    /**
+     * @var array
+     */
+    private $arguments;
+
+    /**
+     * Specify data type for given index. To index JSON you must have the RedisJSON module to be installed.
+     *
+     * @param  string $modifier
+     * @return $this
+     */
+    public function on(string $modifier = 'hash'): self
+    {
+        if (strtoupper($modifier) === 'HASH' || strtoupper($modifier) === 'JSON') {
```

Comment:
I think ```in_array``` is better here. One day search should support more types (streams?) and this would be limiting. WDYT?

Reply:
Yeah, totally agree. P.S Lucking Enums so much 😩